### PR TITLE
fix various model hardcoding. Should move to model features

### PR DESCRIPTION
**Pull request by tarasglek**
Eg we should add functionality to identify model capabilities:
- will it accept image input
- is it speech to text
- is it text to speech

Then we can filter models better and help users decide what to use

### CODE REVIEW

**Comment by tarasglek** [comment]
@humphd i ran prettier, but vs code still do a bunch of header "cleanups" in this diff. wdyt re this change?

**Comment by tarasglek** [comment]
We can do this for vlms too

**Review by humphd — APPROVED** [review]
Looks good, works great

**Comment by humphd** [comment]
@Amnish04 this touches some of your code, in case you want to look too. I think the changes are fine.

File: src/components/Preferences/ModelsSettings.tsx [hunk 1]
```diff
@@ -500,8 +500,9 @@ function ModelsSettings(isOpen: ModelsSettingsProps) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isOpen]);
 
+  // we have multiple isTtsSupported
   const isTtsSupported = useMemo(() => {
-    return !!models.filter((model) => model.id.includes("tts"))?.length;
+    return !!models.find((model) => isTextToSpeechModel(model.id));
   }, [models]);
```
Comment (Amnish04) [inline src/components/Preferences/ModelsSettings.tsx:506]: Yes, this needs to be moved inside a hook. Filed #703 for this.

**Review by Amnish04 — APPROVED** [review]
This looks pretty good! Thanks for cleaning up parts of my code.

**Pull request merged**